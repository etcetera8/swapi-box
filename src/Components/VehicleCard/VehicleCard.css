.VehicleCard {
  background: rgba(50, 255, 250, 0.1);
}