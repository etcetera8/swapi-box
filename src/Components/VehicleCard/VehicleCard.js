import React from 'react';
import PropTypes from 'prop-types';
import './VehicleCard.css'

const VehicleCard = ({vehicle, setFavorite, className}) => {
  const {name, model, passengers, vehicle_class} = vehicle;
  return (
    <article className='Card VehicleCard'> 
      <header>
        <h1 className="name">{name}</h1>
        <button onClick={() => setFavorite(name, 'vehicles')} className={`favorite ${className}`}>â˜…</button>
      </header>
      <ul className="categories">
       <li><span className='key'>Model:</span> {model}</li>
       <li><span className='key'>Vehicle Class:</span> {vehicle_class}</li>
       <li><span className='key'>Passengers:</span> {passengers}</li>
      </ul> 
    </article>
  )
}

VehicleCard.propTypes = {
  vehicle: PropTypes.shape({
    name: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    passengers: PropTypes.string.isRequired,
    vehicle_class: PropTypes.string.isRequired,
  }),
  setFavorite: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired
}

export default VehicleCard;