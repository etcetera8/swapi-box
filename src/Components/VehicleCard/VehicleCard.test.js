import React from 'react';
import ReactDOM from 'react-dom';
import VehicleCard from './VehicleCard';
import { shallow } from 'enzyme';

describe('VehicleCard', () => {
  let wrapper;
  const mockFunction = jest.fn();
  const mockVehicle = {name:'racer', model: 'speedy', passengers:'20', vehicle_class:'old'}
  beforeEach( () => {
    wrapper = shallow(<VehicleCard
                        className={'bike'}
                        setFavorite={mockFunction}
                        vehicle={mockVehicle}
                        key={1}
                      />)
  })

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })
})