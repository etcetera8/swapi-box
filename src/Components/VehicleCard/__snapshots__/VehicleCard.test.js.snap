// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match the snapshot test 1`] = `
<article
  className="Card"
>
  <header>
    <h1
      className="name"
    >
      racer
    </h1>
    <button
      className="favorite bike"
      onClick={[Function]}
    >
      â˜…
    </button>
  </header>
  <ul
    className="categories"
  >
    <li>
      <span
        className="key"
      >
        Model:
      </span>
       
      speedy
    </li>
    <li>
      <span
        className="key"
      >
        Vehicle Class:
      </span>
       
      old
    </li>
    <li>
      <span
        className="key"
      >
        Passengers:
      </span>
       
      20
    </li>
  </ul>
</article>
`;
